Automattic/jetpack
# Check request body error type before creating wp error

Stripe formats its error messages as an array but the Jetpack request is looking for errors as strings.  WCPay already checks whether the response is an error and throws an API_Exception so this PR only ignores errors that are not strings so we don't have any critical errors in `WP_Error`. 

Fixes https://github.com/Automattic/woocommerce-payments/issues/4779#issuecomment-1251972023

#### Changes proposed in this Pull Request:

* Check request body error type and ignore non-string error before creating `WP_Error`

#### Testing instructions:
To reproduce this error, you'll need:
- PHP 8.0+ (since invalid offsets are a warning in PHP 7.4 but an [error in PHP 8.0](https://www.php.net/manual/en/migration80.incompatible.php))
- Install Jetpack 11.3+ on your site
- Create a subscription product with a free trial
- Complete checkout with the test card number `4000000000000002` (generic decline)

<img width="674" alt="Screen Shot 2022-09-19 at 3 57 21 PM" src="https://user-images.githubusercontent.com/56378160/191394825-06f3eda9-8f00-42a7-bf83-e7990cc5f03e.png">

If check out this PR, shouldn't see this error.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?
